lib/pdn/pdn.c

```diff
@@ -598,6 +599,35 @@ int pdn_id_get(uint8_t cid)
 	return strtoul(p + 1, NULL, 10);
 }
 
+int pdn_dynamic_params_get(uint8_t cid, struct in_addr *dns4_pri,
+			   struct in_addr *dns4_sec, unsigned int *ipv4_mtu)
+{
+	int ret;
+	char resp[256];
+	const char *fmt;
+	char dns4_pri_str[INET_ADDRSTRLEN];
+	char dns4_sec_str[INET_ADDRSTRLEN];
+
+	ret = nrf_modem_at_cmd(resp, sizeof(resp), "AT+CGCONTRDP=%u", cid);
+	if (ret) {
+		LOG_ERR("Failed to read dynamic params for CID %u, err %d", cid, ret);
+		return ret;
+	}
+	   /* "+CGCONTRDP: 0,,"example.com","","","198.276.154.230","12.34.56.78",,,,,1464" */
+	fmt = "+CGCONTRDP: %*u,,\"%*[^\"]\",\"\",\"\",\"%15[0-9.]\",\"%15[0-9.]\",,,,,%u";
```

Comment:
If you have a scanf format, why not use `nrf_modem_at_scanf` instead of `nrf_modem_at_cmd`?
Using `nrf_modem_at_scanf` which will scanf directly from shared memory, so you won't need any intermediate buffer on the stack (which in this case is quite big, 256 bytes).